c10/core/DispatchKey.h

```diff
@@ -38,24 +83,19 @@ enum class DispatchKey : uint8_t {
   // this will get eliminated, but for now it's convenient)
   CatchAll = Undefined,
 
-  // ~~~~~~~~~~~~~~~~~~~~~~~~~~ BACKENDS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
-  // A "backend" is colloquially used to refer to handlers for dispatch
-  // which actually implement the numerics of an operation in question.
-  //
-  // Due to the nature of the enum, these backends are specified in
-  // an ordered way, but for most backends this order is not semantically
-  // meaningful (e.g., it's valid to reorder these backends without changing
-  // semantics).  The only situation when backend ordering is meaningful
-  // is when the backend participates in multiple dispatch with another
-  // backend; e.g., CPU and SparseCPU (sparse must have
-  // higher priority).
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~ Functionality Keys ~~~~~~~~~~~~~~~~~~~~~~ //
+  // [Note: Functionality Dispatch Keys]
+  // TODO: fill this in.
+  // See [Note: Per-Backend Functionality Dispatch Keys]
+
+  // See [Note: Per-Backend Functionality Dispatch Keys]
+  Dense,
+
+  // Below are non-extensible backends.
+  // These are backends that currently don't have their own overrides for Autograd/Sparse/Quantized kernels,
+  // and we therefore don't waste space in the runtime operator table allocating space for them.
+  // If any of these backends ever need to customize, e.g., Autograd, then we'll need to add a DispatchKey::*Bit for them.
 
-  // Here are backends which you think of as traditionally specifying
-  // how to implement operations on some device.
-  CPU, // registered at build/aten/src/ATen/RegisterCPU.cpp
-  CUDA, // registered at build/aten/src/ATen/RegisterCUDA.cpp
-  HIP, // NB: I think this is not actually used, due to Note [Masquerading as
```

Comment:
I kept it, it's just moved further down in the enum (https://github.com/pytorch/pytorch/pull/69633/files#diff-8b18203fde3227caea34b6078124836a22aca55e94d3d2b04fe781145adbddbbR327).

Mostly just left it to avoid making extra unnecessary changes in the PR. I'm not... actually sure if the HIP dispatch key is used by anything? I could imagine the Hipifier logic just re-using the CUDA key, but I haven't looked into it too much.

Reply:
Yes, saw that after!